src/components/NostoProduct/store.ts

```diff
@@ -40,41 +49,50 @@ export function createStore(element: NostoProduct) {
   }
 
   function selectSkuId(skuId: string) {
-    state.selectedSkuId = skuId
-    notify("selectedSkuId", skuId)
+    notify("selectedSkuId", (state.selectedSkuId = skuId))
+    const sku = state.skuData?.find(sku => sku.id === skuId)
+    if (sku) {
+      setSkuFields(sku)
+    }
   }
 
   function selectSkuOption(optionId: string, skuIds: string[]) {
     state.skuOptions[optionId] = skuIds
     notify("skuOptions", state.skuOptions)
     const totalSelection = Object.keys(state.skuOptions).length
+    const selectedSkuIds = intersectionOf(...Object.values(state.skuOptions))
 
-    if (totalSelection === state.optionGroupCount) {
-      const selectedSkuIds = intersectionOf(...Object.values(state.skuOptions))
-      if (selectedSkuIds.length === 1) {
+    if (selectedSkuIds.length === 1) {
+      if (totalSelection === state.optionGroupCount) {
         selectSkuId(selectedSkuIds[0])
+      } else if (state.skuData) {
+        const sku = state.skuData.find(sku => sku.id === selectedSkuIds[0])
+        if (sku) {
+          setSkuFields(sku)
+        }
       }
+    } else if (state.skuData) {
+      const skus = state.skuData.filter(sku => selectedSkuIds.includes(sku.id))
```

Comment:
Do we expect more than one SKU object with the same Id?

Reply:
no

Reply:
In that case, can we use find method instead of filter method and update [this](https://github.com/Nosto/web-components/pull/154/files#diff-6276275a0a29c6a17ac3f78858e6470430863c9f09f4de7136668c88085dcde2R143) method to accept single SKU object?

Reply:
I don't get your point. The case for selectedSkuIds.length === 1 is handled starting from line 65
this branch for updating product data when multiple SKUs are selected